### PR TITLE
Include python3.dll in the binary distribution for use in add-ons.

### LINKED ISSUE #15594: Python 3.11: include python3.dll as part of binary distribution
Hi,

Part of #12064 work and subsequent pull requests:

### Background:
Recent Python releases defined limited C API for use by extensions. This allows C extensions to create a .pyd file that will work across Python 3.x releases, including 3.6 and later.

### Is your feature request related to a problem? Please describe.
Suppose an add-on uses external libraries interfacing with CPython runtime (usually .pyd files). If these C extensions are written using limited API, it will allow add-ons to work across Python 3.x releases. At the moment these modules must be accompanied by python3.dll from Python 3.7 installation folder because NVDA itself does not include python3.dll as part of binary distribution.

### Describe the solution you'd like
As part of Python 3.11 upgrade, include python3.dll as part of NVDA binary distribution.

### Describe alternatives you've considered
Let add-ons ship with multiple versions of third-party dependencies, one for each Python version NVDA supports.

### Additional context
Add-ons affected include Resource monitor, Sound Splitter, and others using psutil module.

A bigger problem may arise once 64-bit CPython runtime is used as add-ons must include dependencies supporting both 32-bit and 64-bit Python runtime.

Thanks.

### PR DESCRIPTION
**Pull request by mzanm**
### Link to issue number:
fixes #15594 

### Summary of the issue:
NVDA add-ons that rely on .pyd files, like Resource Monitor, encounter compatibility issues due to the absence of python3.dll in NVDA's distribution. The issue becomes more complicated with NVDA's upgrade to Python 3.11, as it necessitates manually bundling multiple versions of python3.dll to maintain backward compatibility.

### Description of user facing changes
None.

### Description of development approach
In the setup.py file, bundle python3.dll from sys.base_prefix, which should be the python version used to build NVDA.

### Testing strategy:
Override compatibility for Resource Monitor 23.10. It should function normally instead of a DLL error occurring.

### Known issues with pull request:
None.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by mzanm** [comment]
CC @josephsl

**Review by LeonarddeR — APPROVED** [review]

File: source/setup.py [hunk 1]
```diff
@@ -237,7 +237,8 @@ def _genManifestTemplate(shouldHaveUIAccess: bool) -> tuple[int, int, bytes]:
 		("louis/tables",glob("louis/tables/*")),
 		("COMRegistrationFixes", glob("COMRegistrationFixes/*.reg")),
 		(".", glob("../miscDeps/python/*.dll")),
-		(".", ['message.html' ])
+		(".", ['message.html']),
+		(".", [os.path.join(sys.base_prefix, "python3.dll")])
```
Comment (seanbudd) [inline source/setup.py:241]: added comma to minimize future diffs

```suggestion
		(".", [os.path.join(sys.base_prefix, "python3.dll")]),
```

File: user_docs/en/changes.t2t [hunk 2]
```diff
@@ -46,6 +46,7 @@ Please refer to [the developer guide https://www.nvaccess.org/files/nvda/documen
 Add-ons will need to be re-tested and have their manifest updated.
 - Added extension point: ``treeInterceptorHandler.post_browseModeStateChange``. (#14969, @nvdaes)
 - It is now possible to use plural forms in an add-on's translations. (#15661, @beqabeqa473)
+- Included python3.dll in the binary distribution for use by add-ons with external libraries utilizing the stable ABI. (#15674, @mzanm)
```
Comment (seanbudd) [inline user_docs/en/changes.t2t:49]:
```suggestion
- Included python3.dll in the binary distribution for use by add-ons with external libraries utilizing the [stable ABI https://docs.python.org/3.11/c-api/stable.html]. (#15674, @mzanm)
```

**Review by seanbudd — APPROVED** [review]

**Comment by XLTechie** [comment]
@seanbudd For the stable ABI link, why not use the full direct URL?

https://docs.python.org/3.11/c-api/stable.html#stable-abi

**Pull request merged**